src/modules/modals/EditContext.vue

```diff
@@ -136,7 +139,7 @@ export default {
 						id: parseInt(resource.id),
 						type: parseInt(resource.nodeType),
 						// TODO get right permissions for the node
-						permissions: 660,
+						permissions: this.getPermissionBitmaskFromBools(true /* ensure permission is always true */, resource.permissionCreate, resource.permissionUpdate, resource.permissionDelete),
```

Comment:
@blizzz 

- Am I right in assuming that the we should always ensure that a resource has a read permission? Otherwise, there's no point in it being shared?
- When I change the permissions of a resource and update a context using the `Context#update` route, the permission does not update. I believe there might be a bug in the backend code. 
- Why is `permissions` a bitmask, instead of just having multiple boolean variables for the various permissions like is done in other cases? I'm having to  convert from bitmask to boolean anyway (see `getPermissionBitmaskFromBools()` and confirm if that's being done right, please) and wondering if this is avoidable.

Reply:
> Am I right in assuming that the we should always ensure that a resource has a read permission? Otherwise, there's no point in it being shared?

Make sense, yes.

> When I change the permissions of a resource and update a context using the Context#update route, the permission does not update. I believe there might be a bug in the backend code.

Sounds like a bug, yeah. I was testing this stuff always, but cannot exclude it got smuggled in.

> Why is permissions a bitmask, instead of just having multiple boolean variables for the various permissions like is done in other cases?

From backend perspective it is nicer, because in the database you just need one field, and also backend wise it is easy to check whether a bit is sed, and you do not need to deal with a dozen boolean variables at once. This also reduces the parameter count of methods quite a lot :D

Now if it is more straight forward for you to pass on booleans instead, we can do the mapping on the backend instead. IIRC I added methods for this already anyway, to deal with the existing code in PermissionService, if I don't mix up something.